### PR TITLE
Support node injection for bootstrapping

### PR DESCRIPTION
**Pull request by zippy**
This PR adds 

- [x] support for node-injection to allow tests to be decoupled from network bootstrap-server and proxy servers.
- [x] ci and makes explicit which version of holochain it is tested against.

The key addition is a ScenarioApi helper function: `shareAllNodes` which takes a list of `Players`  which will inform all the players in the list about all the other players, thereby removing the need for a bootstrap server and allowing them to communicate on the local network.

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -93,6 +93,25 @@ export class ScenarioApi {
     return players
   }
 
+  shareAllNodes = async (players: Array<Player>)  => {
+    const agentInfos = await Promise.all(players.map(
+      player => player.adminWs().requestAgentInfo({cell_id: null})
+    ))
```
Comment (maackle) [inline src/api.ts:99]: looks like this isn't being used

File: src/api.ts [hunk 2]
```diff
@@ -93,6 +93,25 @@ export class ScenarioApi {
...
+    var player_nodes = {}
```
Comment (maackle) [inline src/api.ts:100]:
```suggestion
    let player_nodes = {}
```

**Review by maackle — COMMENTED** [review]
Well, it makes sense how you hooked it up. My guess would be either that `addAgentInfo` isn't actually working as intended, or that it takes time for the added agents to be recognized.

File: src/api.ts (hunk 1 above)
Comment (zippy) [reply src/api.ts]: yep

**Comment by zippy** [comment]
> Well, it makes sense how you hooked it up. My guess would be either that `addAgentInfo` isn't actually working as intended, or that it takes time for the added agents to be recognized.

Oddly things seem to be working this morning.  Very weird.

File: README.md [hunk 3]
```diff
@@ -114,9 +120,13 @@ const report = await orchestrator.run()
 console.log(report)
 ```
 
-### How to set networking:
+### Networking and tests:
+
+By default tryorama assumes un-bootstrapped `Quic` networking.
```
Comment (Connoropolous) [inline README.md:125]: This is claimed, but I don't see how this is actually true yet, it doesn't appear that a default has been set as such

**Review by Connoropolous — CHANGES_REQUESTED** [review]
I could be wrong about this, but I couldn't see it yet

File: README.md (hunk 3 above)
Comment (zippy) [reply README.md:125]: Hah!  You are right.  I did that in the tests, not in the actual code!

File: package.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
   "author": "",
   "license": "Apache-2.0",
   "dependencies": {
-    "@holochain/conductor-api": "0.0.1-dev.12",
+    "@holochain/conductor-api": "holochain/holochain-conductor-api#dec9231e54f23a69692afb41ba54a8a19688a881",
```
Comment (Connoropolous) [inline package.json:28]: before merging, change this back to the new release of conductor-api

**Review by Connoropolous — APPROVED** [review]
just one todo, but approved

File: README.md [hunk 5]
```diff
@@ -154,6 +164,7 @@ const network = {
 }
 Config.gen({network})
 ```
+Note that in this configuration, if you are using multiple conductors in your tests, they will never be able to see eachother!
```
Comment (Connoropolous) [inline README.md:167]:
```suggestion
Note that in this configuration, if you are using multiple conductors in your tests, they will never be able to see each other!
```

**Review by Connoropolous — COMMENTED** [review]
Minor wording

**Review by zo-el — APPROVED** [review]

File: src/conductor.ts [hunk 6]
```diff
@@ -81,7 +81,8 @@ export class Conductor {
       agent_key: agentPubKey,
       dnas: dnaPaths.map((dnaPath, index) => ({
         path: dnaPath,
-        nick: `${index}${dnaPath}-${uuidGen()}`
+        nick: `${index}${dnaPath}-${uuidGen()}`,
```
Comment (zo-el) [inline src/conductor.ts:84]: Should we update `uuidGen()` to `this._player.scenarioUUID`?

Comment (zippy) [reply src/conductor.ts:84]: I think it should different across each call to installHapp that would keep it constant.

**Pull request merged**